receiver/mongodbreceiver/metrics.go

```diff
@@ -280,161 +250,152 @@ func (s *mongodbScraper) recordCacheOperations(now pcommon.Timestamp, doc bson.M
 		return
 	}
-
-	cursorTimeoutCountValue, err := parseInt(cursorTimeoutCount)
-	if err != nil {
-		errs.AddPartial(1, err)
-	}
-	s.mb.RecordMongodbCursorTimeoutCountDataPoint(now, cursorTimeoutCountValue)
+	s.mb.RecordMongodbCursorTimeoutCountDataPoint(now, val)
 }
 
 func (s *mongodbScraper) recordNetworkCount(now pcommon.Timestamp, doc bson.M, errs *scrapererror.ScrapeErrors) {
-	// Collect network bytes receive, transmit, and request count
 	networkRecorderMap := map[string]func(pcommon.Timestamp, int64){
 		"bytesIn":     s.mb.RecordMongodbNetworkIoReceiveDataPoint,
 		"bytesOut":    s.mb.RecordMongodbNetworkIoTransmitDataPoint,
 		"numRequests": s.mb.RecordMongodbNetworkRequestCountDataPoint,
 	}
 	for networkKey, recorder := range networkRecorderMap {
-		network, err := dig(doc, []string{"network", networkKey})
-		if err != nil {
-			errs.AddPartial(1, err)
-			s.logger.Error("failed to find network", zap.Error(err), zap.String("network", networkKey))
-		}
-		networkValue, err := parseInt(network)
+		metricPath := []string{"network", networkKey}
+		val, err := collectMetric(doc, metricPath)
 		if err != nil {
-			errs.AddPartial(1, err)
-		} else {
-			recorder(now, networkValue)
+			errs.AddPartial(1, fmt.Errorf(collectMetricError, networkKey, err))
+			s.logger.Warn(fmt.Sprintf(collectMetricWarning, networkKey), zap.Error(err))
+			continue
 		}
+		recorder(now, val)
 	}
 }
 
 // Index Stats
 func (s *mongodbScraper) recordIndexAccess(now pcommon.Timestamp, documents []bson.M, dbName string, collectionName string, errs *scrapererror.ScrapeErrors) {
 	// Collect the index access given a collection and database if version is >= 3.2
 	// https://www.mongodb.com/docs/v3.2/reference/operator/aggregation/indexStats/
-	mongo40, _ := version.NewVersion("3.2")
-	if s.mongoVersion.GreaterThanOrEqual(mongo40) {
+	mongo32, _ := version.NewVersion("3.2")
+	if s.mongoVersion.GreaterThanOrEqual(mongo32) {
+		metricName := "mongodb.index.access.count"
 		var indexAccessTotal int64
 		for _, doc := range documents {
+			metricAttributes := fmt.Sprintf("%s, %s", dbName, collectionName)
 			indexAccess, ok := doc["accesses"].(bson.M)["ops"]
 			if !ok {
-				errs.AddPartial(1, errors.New("could not find key for metric"))
+				err := errors.New("could not find key for index access metric")
+				errs.AddPartial(1, fmt.Errorf(collectMetricWithAttributes, metricName, metricAttributes, err))
+				s.logger.Warn(fmt.Sprintf(collectMetricWarningWithAttributes, metricName, metricAttributes), zap.Error(err))
 				return
 			}
 			indexAccessValue, err := parseInt(indexAccess)
 			if err != nil {
-				errs.AddPartial(1, err)
+				errs.AddPartial(1, fmt.Errorf(collectMetricWithAttributes, metricName, metricAttributes, err))
+				s.logger.Warn(fmt.Sprintf(collectMetricWarningWithAttributes, metricName, metricAttributes), zap.Error(err))
 				return
 			}
 			indexAccessTotal += indexAccessValue
 		}
+		if len(documents) == 0 {
```

Comment:
If this needs to be an error, move this check before the loop.

Reply:
not needed check